### ISSUE TITLE
sort list ouput according to date

### ISSUE THREAD

**Issue by asubedi** [pubs/pubs#109]
It would be nice to have the option to sort the output of the list command according to the date.

**Comment by omangin** [comment]
Yes, indeed. For now the output is sorted according to the date of addition to the repository, and there is an argument for sorting alphabetically according to the citekey. Before adding this we might want to make it a little bit more general and have the ability to sort according to various other fields (eventually a list of them following lexicographic order). Does anyone have suggestions about what we would want to include. I am thinking of the following obvious ones (also including a reverse option for each):
– *citekey*: citekey alphabetically,
– *added*: date of addition to the repository,
– *authors*: authors last names (lexicographic when several authors),
– *date*: date of publication.

The argument would be something like: `--order date,citekey`. Any comment, suggestion on that? Which would be the most useful in your opinion?

**Comment by wflynny** [comment]
Agreed.  I think the following options should be included:

    -r, --reverse
    -o, --order field1[,field2,...]

so that these use cases can be supported:

    pubs list -r
    pubs list -ir -o author,date

It would also be nice if some options have aliases that resolve to the same filter.  Such as `date` and `year` both resolving to sorting by date of publication.

**Comment by wflynny** [comment]
I'd also argue if changes are coming to `list_cmd`, then citekey justification on listing should be added too (e.g. adding some string formatting to `pretty.py: paper_oneliner`), though this should probably be in a separate issue.  Something like: `u'[{citekey:>30}] {descr} {tags}'`

**Comment by omangin** [comment]
I created a separate issue for the citekey indentation.

Any volunteer for implementing the sorting option?
